stages/org.osbuild.grub2.inst

```diff
@@ -136,15 +141,23 @@ def prefix_partition(options: Dict):
 def main(tree, options):
     filename = options["filename"]
     platform = options["platform"]
-    location = options["location"]
     sector_size = options.get("sector-size", 512)
 
     image = os.path.join(tree, filename.lstrip("/"))
 
-    prefix = prefix_partition(options["prefix"])
+    if "number" in options["prefix"]:
+        prefix = prefix_partition(options["prefix"])
+    else:
+        prefix = options["prefix"]["path"]
     print(f"prefix: {prefix}")
     core_path = core_mkimage(platform, prefix, options["core"])
 
+    # If location isn't set, use the image file as-is instead of with the MBR
+    if "location" not in options:
+        shutil.copyfile(core_path, image)
```

Comment:
(super nitpick) I wonder if extracting the code below "location = ..\nwith openIimage" into a helper and calling that helper if "location" is in options instead of the early return. Its maybe a bit far-fetched but as code grows maybe someone adds some new feature/option at the end of new and then we would have a very subtle bug that some feature only works if "location" is set (because otherwise we have this early return). But I admit it's quite theoretical :)

Reply:
Yeah, that makes main nicer.